presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -311,6 +311,36 @@ public void testGroupByWithSubquerySelectExpressionWithDereferenceExpression()
     }
 
     @Test
+    public void testGroupByWithSubquerySelectExpressionWithLambda()
+    {
+        // GROUP BY column captured in lambda
+        analyze(
+                "SELECT (SELECT apply(0, x -> x + b) from (values 1) x(a)) " +
+                        "FROM t1 u GROUP BY b");
+
+        // non-GROUP BY column captured in lambda
+        assertFails(
+                MUST_BE_AGGREGATE_OR_GROUP_BY,
+                "line 1:34: Subquery uses '\"a\"' which must appear in GROUP BY clause",
+                "SELECT (SELECT apply(0, x -> x + a) from (values 1) x(c)) " +
+                        "FROM t1 u GROUP BY b");
+        // TODO #7784
+//        assertFails(
+//                MUST_BE_AGGREGATE_OR_GROUP_BY,
+//                "line 1:34: Subquery uses '\"u.a\"' which must appear in GROUP BY clause",
+//                "SELECT (SELECT apply(0, x -> x + u.a) from (values 1) x(a)) " +
+//                        "FROM t1 u GROUP BY b");
+
+        // name shadowing
+        analyze(
```

Comment:
Correlated subqueries are not yet fully supported by the execution. This PR fixes analysis by rejecting invalid queries, but supporting valid ones during execution requires more work.

I wanted to add tests:
```
assertQuery("SELECT (SELECT a) FROM (VALUES 1, 2) t(a)", "VALUES (1), (2)");
assertQuery("SELECT (SELECT apply(1, x -> x + a)) FROM (VALUES 1, 2) t(a)", "VALUES (2), (3)");
assertQuery("SELECT a FROM (VALUES 1, 2) t(a) ORDER BY (SELECT apply(1, x -> x - a))", "VALUES (2), (1)");
```

(even the first one -- no lambdas, no aggregations -- fails now)